app/models/tag.rb

```diff
@@ -22,20 +22,25 @@ class Tag < ApplicationRecord
   before_save :normalize_name
 
   # -- Validations ------------------------------------------------------------
-  validates :name, presence: true, uniqueness: { case_sensitive: false }
+  validates :name,
+  presence: true,
+  uniqueness: { case_sensitive: false } ,
+  format: {
+    with: /\A(!\h{6})_[a-zA-Z]+?\z/, message: 'is invalid: Numbers and special characters are not permitted.'
+  }
 
   # -- Scopes -----------------------------------------------------------------
 
-
   # -- Class Methods ----------------------------------------------------------
 
-
   # -- Instance Methods -------------------------------------------------------
 
   # Returns a version of the tag's name suitable to present to the user:
   #  * If the tag name contains color details, they are stripped
   #  * The result is titleized
   def display_name()
+    return '' if self.name.nil?
```

Comment:
This seems like a validation failure? Under what conditions would we be having a `#display_name` call, on a Tag that has a `nil` name?

Reply:
We needed this condition for the tag form when there is a new record of the tag. In tag form when `#display_name` or `#color` fields try to retrieve its value from `Tag.new` have the name value of nil we get an error of `nil:NilClass`. In order to prevent that error from happening we added a nil guard. 

I have refactored my implementation, now we are checking if the @tag object is new_record? then using `#display_name` or `#color` to retrieve the values if that's not the case then we are setting empty string('') as the value for these fields.